Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator.hs

```diff
@@ -38,6 +40,17 @@ instance JobProcessor AllocatorJobType where
   restoreAnyJobInfo SSendPaymentReminderToDriver jobData = AnyJobInfo <$> restoreJobInfo SSendPaymentReminderToDriver jobData
   restoreAnyJobInfo SUnsubscribeDriverForPaymentOverdue jobData = AnyJobInfo <$> restoreJobInfo SUnsubscribeDriverForPaymentOverdue jobData
   restoreAnyJobInfo SUnblockDriver jobData = AnyJobInfo <$> restoreJobInfo SUnblockDriver jobData
+  restoreAnyJobInfo SEndRideAfterThresholdTimePassed jobData = AnyJobInfo <$> restoreJobInfo SEndRideAfterThresholdTimePassed jobData
+
+data EndRideAfterThresholdTimePassedJobData = EndRideAfterThresholdTimePassedJobData
+  { rideId :: Id Ride,
+    merchantId :: Id DM.Merchant
```

Comment:
Alternative : We could have made it as one entry in scheduler which retries itself each x minutes/hours. The scheduler could pick ALL rides which were stuck at that moment. 
With the current implementation :
1. If the job fails for a ride(due to any error etc), it won't be picked in the next job
2. Too many job entries in scheduler table